engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/CatchPolyglotSymbolTypeBranchNode.java

```diff
@@ -0,0 +1,75 @@
+package org.enso.interpreter.node.controlflow.caseexpr;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.node.expression.builtin.meta.IsSameObjectNode;
+import org.enso.interpreter.node.expression.builtin.meta.TypeOfNode;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.builtin.Builtins;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+import org.enso.interpreter.runtime.data.Type;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+
+/** An implementation of the case expression specialised to working on polyglot types. */
+@NodeInfo(shortName = "PolyglotSymbolTypeMatch")
+public abstract class CatchPolyglotSymbolTypeBranchNode extends BranchNode {
+
+  private final Object polyglotSymbol;
+  private @Child TypeOfNode typeOfNode = TypeOfNode.build();
+  private @Child IsSameObjectNode isSameObject = IsSameObjectNode.build();
+  private final ConditionProfile profile = ConditionProfile.createCountingProfile();
+  private final ConditionProfile subtypeProfile = ConditionProfile.createCountingProfile();
+
+  CatchPolyglotSymbolTypeBranchNode(Object polyglotSymbolName, RootCallTarget functionNode) {
+    super(functionNode);
+    this.polyglotSymbol = polyglotSymbolName;
```

Comment:
Once this is called `polyglotSymbol` and other-time it is `polyglotSymbolName` - so is it _the symbol_ or its _name_ after all? I think we should be consistent.